### ISSUE TITLE
investigate how to get accurate estimate of visitors

### ISSUE THREAD

**Issue by hannesdatta** [tilburgsciencehub/website#907]
Since end of august, we do not have proper data on the number of visitors on our site.

Here is what I get from our marketing team from Tilburg University. 

> Voor het plaatsen van analytische cookies hoef je volgens de website geen toestemming te vragen, al moet je websitebezoekers wel informeren dat je deze cookies plaatst: https://autoriteitpersoonsgegevens.nl/themas/internet-slimme-apparaten/cookies
> 
> Sinds we in juli voor advertentiedoeleinden onze GA4 property met Google Ads en Google Signals hebben gekoppeld, valt GA4 voor [www.tilburguniversity.edu](http://www.tilburguniversity.edu/) onder de tracking cookies en mogen we deze cookies niet meer zetten zonder hier toestemming voor te vragen: https://autoriteitpersoonsgegevens.nl/themas/internet-slimme-apparaten/cookies/tracking-cookies
> 
> Dat wil niet zeggen dat we in GA4 helemaal geen data hebben van de bezoekers die de cookies weigeren. Google heeft met Consent mode in Google Tag Manager een oplossing om op basis van pings en behavioral modeling een inschatting te geven van de gemiste data: https://support.google.com/analytics/answer/11161109
> 
> Om hiervoor in aanmerking te komen, heb je wel redelijk wat verkeer nodig.

Is this a solution to get the number of visitors back?

Please investigate.

IF NOT, we need a solution asap to know who is visiting our site. Please come up with implementable suggestions in that case.

**Comment by thierrylahaije** [comment]
This will not be the solution, since we do not have enough data:

> - De property verzamelt ten minste 7 dagen lang minimaal 1000 gebeurtenissen met analytics_storage='denied'.
> - De property heeft ten minste 1000 dagelijkse gebruikers die gebeurtenissen sturen met analytics_storage='granted' gedurende ten minste 7 van de afgelopen 28 dagen. 

So we need at least 1000 visitors a day for this model to work... Maybe a solution for later, but for now we will need to keep on investigating for alternatives.

**Comment by hannesdatta** [comment]
Approach discussed w/ @kleintob. Please come up with implementable solutions.

**Comment by thierrylahaije** [comment]
@hannesdatta @kleintob I checked the possibilities in the link sent by Tobi and I also checked out Reddit and Google. We actually have two possibilities, which you should check out:

1. Posthog:

**Features & benefits**
An all-in-one analytics suite
EU data hosting available
Feature flags, A/B testing, heatmaps, session recording and more
Apps to integrate with data warehouses
Unlimited ability to scale
Open source, via MIT license

**PostHog and GDPR compliance**
Open Source: ✔
Self Hosting: ✔
EU Cloud Hosting: ✔
Cookieless Tracking: ✔

2. Plausible.io

**Features & benefits**
Lightweight script with minimal page speed impact
No need for any cookies, at all
Minimal data collection for users
No tracking across sessions, devices or sites
API Possibility

**Plausible and GDPR compliance**
Open Source: ✔
Self Hosting: ✔
EU Cloud Hosting: ✔
Cookieless Tracking: ✔

Plausible costs about 9-19 dollar a month depending on the amount of pageviews. Posthog seems free up to a certain amount of pageviews. Plausible gives you the ability to enable API usage which could be very beneficial for data analysis.

Please let me know what you think!

**Comment by kleintob** [comment]
Thanks! Could you briefly explain why the other options on that website that I shared would not work or would be worse?

For instance, why would Matomo not work? https://matomo.org/pricing/

What’s the simplest solution that would not cost anything, would be GDPR proof, and would give us some simple stats?

From: thierrylahaye ***@***.***>
Date: Tuesday, 24 October 2023 at 01:35
To: tilburgsciencehub/website ***@***.***>
Cc: Tobias Klein ***@***.***>, Mention ***@***.***>
Subject: Re: [tilburgsciencehub/website] investigate how to get accurate estimate of visitors (Issue #907)

@hannesdatta<https://github.com/hannesdatta> @kleintob<https://github.com/kleintob> I checked the possibilities in the link sent by Tobi and I also checked out Reddit and Google. We actually have two possibilities, which you should check out:

  1.  Posthog:

Features & benefits
An all-in-one analytics suite
EU data hosting available
Feature flags, A/B testing, heatmaps, session recording and more
Apps to integrate with data warehouses
Unlimited ability to scale
Open source, via MIT license

PostHog and GDPR compliance
Open Source: ✔
Self Hosting: ✔
EU Cloud Hosting: ✔
Cookieless Tracking: ✔

  1.  Plausible.io

Features & benefits
Lightweight script with minimal page speed impact
No need for any cookies, at all
Minimal data collection for users
No tracking across sessions, devices or sites
API Possibility

Plausible and GDPR compliance
Open Source: ✔
Self Hosting: ✔
EU Cloud Hosting: ✔
Cookieless Tracking: ✔

Plausible costs about 9-19 dollar a month depending on the amount of pageviews. Posthog seems free up to a certain amount of pageviews. Plausible gives you the ability to enable API usage which could be very beneficial for data analysis.

Please let me know what you think!

**Comment by thierrylahaije** [comment]
@kleintob Sorry, I did not take a look at Matomo. It is also a possibility, but the free version is limited to basic web stats. A/B testing and other data analysis functionalities that the others provide for a low price, cost about 150 per month at Matomo. Furthermore, it seems that when you want to use Matomo without cookies, this does reduce the quality of data it collects. 

All in all, I would not focus to much on the pricing issue, but more on the future of the website. We already saw that a lot of users do not click the accept cookies button in the banner (since we have almost no data). Posthog and plausible.io still gives us the ability to do data analysis for a very low pricing, while Matomo can only do this to a minimum extent.

Currently, we almost don't do any kind of data analysis for our website. If the goal of TSH is to grow, we need to do data analysis at the website level to improve the usage of the website. Web data analysis can have a great impact on the usage of a website (as I know from experience with my own company). Therefore I would advice to choose Plausible or Posthog. The pricing is low and the effect of these analytics can have a great impact on TSH. Probably a better (cost-wise) impact than we could achieve using advertising or anything like that.
